packages/puppeteer-core/src/common/util.ts

```diff
@@ -463,6 +464,19 @@ export function fromEmitterEvent<
   });
 }
 
+/**
+ * @internal
+ */
+export function fromAbortSignal(signal?: AbortSignal): Observable<never> {
+  return signal
+    ? fromEvent(signal, 'abort').pipe(
+        map(() => {
+          throw new Error(`Aborted reason: ${signal.reason}`);
```

Comment:
I forgot how we do it in other places but can we use singal.throwIfAborted() here. I think in other places we do not construct custom errors